pallets/gear/src/lib.rs

```diff
@@ -1673,6 +1659,58 @@ pub mod pallet {
 
             Ok(())
         }
+
+        /// Transfers value from terminated or exited program to its inheritor.
+        ///
+        /// `depth` parameter is how far to traverse to inheritor.
+        #[pallet::call_index(8)]
+        #[pallet::weight(<T as Config>::WeightInfo::claim_value_to_inheritor(depth.get()))]
```

Comment:
Correct me if I'm wrong - user will be charged by the _depth.get()_ and compensated later for the actually used gas. Is that right?

So the guideline usage for the `depth` is to set it to `MAX`. Right?
I will explain why I'm asking that. Actually, this extrinsic is a great idea. Depth param gives control over funds in 2 ways:
1. You can hack inheritors cycles (read other comments)
2. You can collect funds only from chosen holders.
But to use these great opportunities one should know the correct `depth` param. I can't see any easy way to understand what is the actual depth or what is the desired depth for user. Are there any ways to get this info?

Reply:
In case of MAX the expected weight of extrinsic would exceed maximum allowed for DispatchType::Normal, so it's better to set something reasonable. For most of cases depth will be eq 1

Reply:
Moreover, can't see refund here @ark0f 
Please collect amount of visited accounts and return `Ok(WeightInfo::claim_value_to_inheritor(real_depth).into())` instead of `Ok(().into())`

Reply:
> Moreover, can't see refund here @ark0f Please collect amount of visited accounts and return `Ok(WeightInfo::claim_value_to_inheritor(real_depth).into())` instead of `Ok(().into())`

Great that you have mentioned this, as all the tests actually use `MAX`. That would be a wrong guideline for the users